### PR TITLE
Replace HEXENCODEDHASH with hash

### PR DESCRIPTION
**Pull request by batconjurer**
Hoping to convince you this simplifies the code :sweat_smile:

### CODE REVIEW

**Comment by sug0** [comment]
hash clash. LOL

File: apps/src/lib/node/ledger/events/log/dumb_queries.rs [hunk 1]
```diff
@@ -6,46 +6,59 @@
...
+    pub fn matches(&self, event: &'a Event) -> bool {
         event.event_type == self.event_type
             && event
-                .attributes
-                .get(&self.attr)
-                .map(|value| value == self.value)
-                .unwrap_or_default()
+            .attributes
+            .get(&self.attr)
+            .map(|value| T::try_from(value.as_ref()).map(|v| v == self.value).unwrap_or(false))
```
Comment (sug0) [inline apps/src/lib/node/ledger/events/log/dumb_queries.rs:38]: see this is the part that bothers me: to do these kind of checks, you need to decode the hex encoded string into a `Hash`, for every iteration over the event log

File: apps/src/lib/node/ledger/events/log.rs [hunk 2]
```diff
@@ -67,13 +67,12 @@ impl EventLog {
...
-        &'log self,
-        matcher: dumb_queries::QueryMatcher<'query>,
-    ) -> impl Iterator<Item = &'log Event> + 'query
+    pub fn iter_with_matcher<'a, T>(
+        &'a self,
+        matcher: dumb_queries::QueryMatcher<'a, T>,
+    ) -> impl Iterator<Item = &Event> + 'a
     where
-        // the log should outlive the query
-        'log: 'query,
```
Comment (sug0) [inline apps/src/lib/node/ledger/events/log.rs:76]: we do get rid of this bit of lifetime gymnastics though, which is nice to the reader

File: shared/src/types/hash.rs [hunk 3]
```diff
@@ -95,6 +92,26 @@ impl TryFrom<&[u8]> for Hash {
     }
 }
 
+impl TryFrom<String> for Hash {
+    type Error = self::Error;
+
+    fn try_from(string: String) -> HashResult<Self> {
+        Ok(
+            Self(<[u8; HASH_LENGTH]>::from_hex(string).map_err(Error::FromStringError)?)
+        )
```
Comment (sug0) [inline shared/src/types/hash.rs:101]: here you could have used `try_into()` to avoid copy and pasting the same def

Comment (sug0) [reply shared/src/types/hash.rs:101]: `string.as_str().try_into()`

File: apps/src/lib/node/ledger/events/log/dumb_queries.rs [hunk 4]
```diff
@@ -6,46 +6,59 @@
...
 use crate::node::ledger::events::{Event, EventType};
 
 /// A [`QueryMatcher`] verifies if a Namada event matches a
 /// given Tendermint query.
 #[derive(Debug, Clone)]
-pub struct QueryMatcher<'q> {
+pub struct QueryMatcher<'a, T>
+where
+    T: 'a +  PartialEq + TryFrom<&'a str>
+{
```
Comment (sug0) [inline apps/src/lib/node/ledger/events/log/dumb_queries.rs:20]: I think it's cleaner to leave trait bounds to impl blocks

Comment (sug0) [reply apps/src/lib/node/ledger/events/log/dumb_queries.rs:20]: i.e. remove the trait bounds here

File: apps/src/lib/node/ledger/events/log/dumb_queries.rs (hunk 1 above)
Comment (batconjurer) [reply apps/src/lib/node/ledger/events/log/dumb_queries.rs:38]: Yeah, but had to do a utf8 conversion before instead.

Comment (sug0) [reply apps/src/lib/node/ledger/events/log/dumb_queries.rs:38]: https://doc.rust-lang.org/stable/src/core/str/converts.rs.html#170-174
it just does a pointer cast, basically

File: shared/src/types/hash.rs (hunk 3 above)
Comment (batconjurer) [reply shared/src/types/hash.rs:101]: true. Copy pasta game weeeeaaak

File: apps/src/lib/node/ledger/events/log/dumb_queries.rs (hunk 4 above)
Comment (batconjurer) [reply apps/src/lib/node/ledger/events/log/dumb_queries.rs:20]: Fair enough. Can literally just remove that and the code will work.

File: apps/src/lib/node/ledger/events/log/dumb_queries.rs (hunk 1 above)
Comment (batconjurer) [reply apps/src/lib/node/ledger/events/log/dumb_queries.rs:38]: that's true because you were using the unchecked version.

**Comment by sug0** [comment]
Merged here 9e7858732f8e02a77b8ac7924ff0f67add7ff542

**Pull request closed without merging**